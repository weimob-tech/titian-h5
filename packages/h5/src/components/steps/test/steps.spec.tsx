import { newSpecPage } from '@stencil/core/testing';
import { h } from '@stencil/core';
import { TiSteps } from '../index';
const options = [
  {
    title: '标题文字',
    subtitle: '副标题',
    description: '详细内容文字，详细内容文字，详细内容文字',
    time: '2018.07.06 09:52:42',
  },
  {
    title: '标题文字',
    subtitle: '副标题',
    description: '详细内容文字，详细内容文字，详细内容文字',
    time: '2018.07.06 09:52:42',
  },
];
describe('渲染 ti-steps', () => {
  it('基本用法', async () => {
    const page = await newSpecPage({
      components: [TiSteps],
      template: () => <ti-steps options={options} current={0} />,
    });
    expect(page.root).toMatchSnapshot();
  });
  it('设置副标题右对齐', async () => {
    const page = await newSpecPage({
      components: [TiSteps],
      template: () => <ti-steps options={options} current={0} subtitle-align="right" />,
    });
    expect(page.root).toMatchSnapshot();
  });
});
