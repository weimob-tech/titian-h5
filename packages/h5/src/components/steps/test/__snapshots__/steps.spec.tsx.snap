// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`渲染 ti-steps 基本用法 1`] = `
<ti-steps>
  <mock:shadow-root>
    <div class="titian-steps undefined">
      <div class="titian-steps-item titian-steps-item-active">
        <div class="titian-steps-item-left">
          <div class="titian-steps-item-icon-box">
            <div class="titian-steps-item-dot"></div>
          </div>
          <div class="titian-steps-item-line"></div>
        </div>
        <div class="titian-steps-item-content">
          <div class="titian-steps-item-title-box">
            <div class="titian-steps-item-title">
              标题文字
            </div>
            <div class="titian-steps-item-subtitle">
              副标题
            </div>
          </div>
          <div class="titian-steps-item-desc">
            详细内容文字，详细内容文字，详细内容文字
          </div>
          <div class="titian-steps-item-time">
            2018.07.06 09:52:42
          </div>
        </div>
      </div>
      <div class="titian-steps-item">
        <div class="titian-steps-item-left">
          <div class="titian-steps-item-icon-box">
            <div class="titian-steps-item-dot"></div>
          </div>
        </div>
        <div class="titian-steps-item-content">
          <div class="titian-steps-item-title-box">
            <div class="titian-steps-item-title">
              标题文字
            </div>
            <div class="titian-steps-item-subtitle">
              副标题
            </div>
          </div>
          <div class="titian-steps-item-desc">
            详细内容文字，详细内容文字，详细内容文字
          </div>
          <div class="titian-steps-item-time">
            2018.07.06 09:52:42
          </div>
        </div>
      </div>
    </div>
  </mock:shadow-root>
</ti-steps>
`;

exports[`渲染 ti-steps 设置副标题右对齐 1`] = `
<ti-steps subtitle-align="right">
  <mock:shadow-root>
    <div class="titian-steps undefined">
      <div class="titian-steps-item titian-steps-item-active">
        <div class="titian-steps-item-left">
          <div class="titian-steps-item-icon-box">
            <div class="titian-steps-item-dot"></div>
          </div>
          <div class="titian-steps-item-line"></div>
        </div>
        <div class="titian-steps-item-content">
          <div class="titian-steps-item-title-box">
            <div class="titian-steps-item-title">
              标题文字
            </div>
            <div class="titian-steps-item-subtitle titian-steps-item-subtitle-right">
              副标题
            </div>
          </div>
          <div class="titian-steps-item-desc">
            详细内容文字，详细内容文字，详细内容文字
          </div>
          <div class="titian-steps-item-time">
            2018.07.06 09:52:42
          </div>
        </div>
      </div>
      <div class="titian-steps-item">
        <div class="titian-steps-item-left">
          <div class="titian-steps-item-icon-box">
            <div class="titian-steps-item-dot"></div>
          </div>
        </div>
        <div class="titian-steps-item-content">
          <div class="titian-steps-item-title-box">
            <div class="titian-steps-item-title">
              标题文字
            </div>
            <div class="titian-steps-item-subtitle titian-steps-item-subtitle-right">
              副标题
            </div>
          </div>
          <div class="titian-steps-item-desc">
            详细内容文字，详细内容文字，详细内容文字
          </div>
          <div class="titian-steps-item-time">
            2018.07.06 09:52:42
          </div>
        </div>
      </div>
    </div>
  </mock:shadow-root>
</ti-steps>
`;
